src/LinuxTracing/ThreadStateManagerTest.cpp

```diff
@@ -96,19 +108,26 @@ TEST(ThreadStateManager, NewTask) {
 TEST(ThreadStateManager, TwoThreads) {
   constexpr pid_t kTid1 = 42;
   constexpr pid_t kTid2 = 52;
+  constexpr pid_t kWasBlockedByTid1 = 420;
+  constexpr pid_t kWasBlockedByPid1 = 4200;
+  constexpr pid_t kWasBlockedByTid2 = 520;
+  constexpr pid_t kWasBlockedByPid2 = 5200;
   ThreadStateManager manager;
   std::optional<ThreadStateSlice> slice;
 
-  manager.OnInitialState(100, kTid1, ThreadStateSlice::kRunnable);
+  manager.OnInitialState(100, kTid1, ThreadStateSlice::kRunnable, kWasBlockedByTid1,
```

Comment:
Again, I don't understand why OnInitialState include this 2 fields.

Reply:
Same as above. Since it is in runnable state it should have blocker tid and pid.

Reply:
I'm with @karupayun here. As I wrote in the other comment, I think these values are never available as "initial state". As a matter of fact, they are only specified in tests.

Reply:
I thought OnInitialState is being used to create a new state that could be runnable, running, or sleeping. In that case, if you use OnInitialState to create a runnable slice and you have the info for what was blocking that state before you would want to add it in OnInitialState.

Reply:
`OnInitialState` is used with the state read from `/proc/[pid]/stat`. See `TracerImpl::RetrieveInitialThreadStatesOfTarget` and `SwitchesStatesNamesVisitor::ProcessInitialState`. You don't have the info on what thread switched this thread to runnable from that. That's why you gave default values to those parameters in the first place.